### ISSUE TITLE
Missing \protected@file@percent

### ISSUE THREAD

**Issue by Rimole** [latex3/babel#224]
babel has

\def\babel@aux#1#2{%
  \select@language{#1}%
  \bbl@foreach\BabelContentsFiles{% \relax -> don't assume vertical mode
    \@writefile{##1}{\babel@toc{#1}{#2}\relax}}}% TODO - plain?

which should be

\def\babel@aux#1#2{%
  \select@language{#1}%
  \bbl@foreach\BabelContentsFiles{% \relax -> don't assume vertical mode
    \@writefile{##1}{\babel@toc{#1}{#2}\relax\protected@file@percent}}}% TODO - plain?

because it is writing to the table of contents (see "\protected@file@percent" in source2e.pdf).
Thank you!

**Comment by jbezos** [comment]
Please, post a MWE showing an actual issue. In principle, a `%` after `\relax` is unnecessary, because with the latter spaces and eols are gobbled. On the other hand, perhaps it would be sensible to *replace* `\relax` by `\protected@file@percent`.

**Comment by FrankMittelbach** [comment]
If I remember correctly I used `\protected@file@percent` back then for two reasons:
- to ensure not to get spaces if in hmode (that could be done equally with `\relax`)
- but also so that an entry has a chance to look at the next without the need to account for a `\relax` or not
But my memory is hazy ... could be that there is more documentation in the relevant part of the source (actually I think it had something to do how titletoc was behaving :-)

**Comment by u-fischer** [comment]
@FrankMittelbach  yes, we added a real % to avoid troubles with look aheads, see egreg comment here:  https://github.com/latex3/latex2e/issues/73

**Comment by jbezos** [comment]
In this case the lookahead is not relevant, because `\bbl@toc` is a `babel` internal macro, which relies on the `\relax` trick. I’ve have doubts there is an actual issue, but who knows.

**Comment by jbezos** [comment]
I’m closing this issue because it isn’t an actual issue.
